File: snippets/dart_snippets/lib/buy_btc.dart
import 'package:breez_sdk/bridge_generated.dart';
import 'package:dart_snippets/sdk_instance.dart';

Future<BuyBitcoinResponse> buyBitcoin() async {
  // ANCHOR: buy-btc
  BuyBitcoinRequest req = const BuyBitcoinRequest(provider: BuyBitcoinProvider.Moonpay);
  BuyBitcoinResponse resp = await breezSDK.buyBitcoin(req: req);
  // ANCHOR_END: buy-btc
  return resp;
}

File: snippets/swift/BreezSDKExamples/Sources/BuyBtc.swift
//
//  BuyBtc.swift
//
//
//  Created by ruben on 14/11/2023.
//

import BreezSDK
import Foundation

func buy(sdk: BlockingBreezServices) -> BuyBitcoinResponse? {
    // ANCHOR: buy-btc
    let buyBitcoinResponse = try? sdk.buyBitcoin(
        req: BuyBitcoinRequest(provider: .moonpay))
    // ANCHOR_END: buy-btc
    return buyBitcoinResponse
}
